### PR TITLE
SPAPI-43 expose a developer message in the api

### PR DESCRIPTION
**Pull request by Kareeeeem**

### CODE REVIEW

File: spynl/main/exceptions.py [hunk 1]
```diff
@@ -16,9 +16,13 @@ class SpynlException(Exception):
     """
     http_escalate_as = HTTPBadRequest
 
-    def __init__(self, message='an internal error has occured', debug_message=None):
+    def __init__(self,
+                 message='an internal error has occured',
+                 debug_message=None,
+                 developer_message=None):
         self.message = message
```
Comment (RosanneZe) [inline spynl/main/exceptions.py]: I think here we need a few short comments:
```python
# friendly message for the end user, exposed, should not contain sensitive information
self.message = message
# debug message for internal use
self.debug_message = debug_message or self.message
# developer message for external programmers that use the Spynl API, exposed, should not contain sensitive information
self.developer_message = developer_message or self.message
```

**Review by RosanneZe — CHANGES_REQUESTED** [review]
Looks good, but because we now have quite a few messages, I think we should be very explicit about what they are for. Maybe we can even repeat some of this info in make_response, but that might be a bit too much?

File: spynl/main/exceptions.py (hunk 1 above)
Comment (ev-agelos) [reply spynl/main/exceptions.py]: Sure but I would put them in the docstring of the __init__ method.
Other than that PR seems fine by me.

**Review by ev-agelos — APPROVED** [review]

**Review by o-m-n-i — COMMENTED** [review]
I think exposing a developer friendly message is nice, not sure about the debug message though. You could accidentally expose sensitive information if somehow the debug flag gets set in a production environment. Curious to know what @koraa thinks.

File: spynl/main/exceptions.py (hunk 1 above)
Comment (Kareeeeem) [reply spynl/main/exceptions.py]: Will put them in the docstring.

**Comment by Kareeeeem** [comment]
The debug message was there before. It is also never sent in the response, it is used for logging. https://github.com/SoftwearDevelopment/spynl/blob/kareem/SPAPI-43/spynl/main/utils.py#L472

**Comment by koraa** [comment]
@Kareeeeem Precisely

**Comment by o-m-n-i** [comment]
@Kareeeeem okay perfect. Then I'm fine with the changes +1

**Review by o-m-n-i — APPROVED** [review]

**Comment by koraa** [comment]
I just looked over the code again, and I am wondering: Do we have any 3pty users using our API?

**Review by RosanneZe — APPROVED** [review]

**Comment by Kareeeeem** [comment]
forgot to force push before deleting the branch. it is merged.

**Comment by Kareeeeem** [comment]
@koraa there will be soon. I remember @bobbydams telling us about a meeting he was in last week that was about 3rd party users.

**Pull request closed without merging**